src/progress.rs

```diff
@@ -50,6 +52,117 @@ impl<'a> Progress<'a> {
+    /// #    fn get_uid(&self) -> &Self::UIDType {
+    /// #        &self.id
+    /// #    }
+    /// # }
+    /// #
+    /// # futures::executor::block_on(async move {
+    /// let client = Client::new("http://localhost:7700", "masterKey");
+    /// let movies = client.create_index("movies_wait_for_pending", None).await.unwrap();
+    /// 
+    /// let progress = movies.add_documents(&[
+    ///     Document { id: 0, kind: "title".into(), value: "The Social Network".to_string() },
+    ///     Document { id: 1, kind: "title".into(), value: "Harry Potter and the Sorcerer's Stone".to_string() },
+    /// ], None).await.unwrap();
+    /// 
+    /// let status = progress.wait_for_pending_update(None, None).await.unwrap();
+    /// 
+    /// # client.delete_index("movies_wait_for_pending").await.unwrap();
+    /// assert!(matches!(status.unwrap(), UpdateStatus::Processed { .. }));
+    /// # });
+    /// ```
+    pub async fn wait_for_pending_update(
+        &self,
+        interval_ms: Option<Duration>,
+        timeout_ms: Option<Duration>,
+    ) -> Option<Result<UpdateStatus, Error>> {
+        let interval: Duration;
+        let timeout: Duration;
+
+        match interval_ms {
+            Some(v) => interval = v,
+            None => interval = Duration::from_millis(50),
+        }
+
+        match timeout_ms {
+            Some(v) => timeout = v,
+            None => timeout = Duration::from_millis(5000),
+        }
+
+        let mut elapsed_time = Duration::new(0, 0);
+        let mut status_result: Result<UpdateStatus, Error>;
+
+        while timeout > elapsed_time {
+            status_result = self.get_status().await;
+
+            match status_result {
+                Ok (status) => {
+                    match status {
+                        UpdateStatus::Failed { .. } | UpdateStatus::Processed { .. } => {
+                            return Some(self.get_status().await);
+                        },
+                        UpdateStatus::Enqueued { .. } => {
+                            elapsed_time += interval;
+                            async_sleep(interval).await;
+                        },
+                    }
+                },
+                Err (error) => return Some(Err(error)),
+            };
+        }
+
+        None
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub(crate) async fn async_sleep(interval: Duration) {
+    let (sender, receiver) = futures::channel::oneshot::channel::<()>();
+    std::thread::spawn(move || {
+        std::thread::sleep(interval);
+        let _ = sender.send(());
+    });
+    let _ = receiver.await;
+}
+
+#[cfg(target_arch = "wasm32")]
+pub(crate) async fn async_sleep(interval: Duration) {
+    use wasm_bindgen_futures::JsFuture;
+
+    JsFuture::from(js_sys::Promise::new(&mut |yes, _| {
+        web_sys::window()
+            .unwrap()
+            .set_timeout_with_callback_and_timeout_and_arguments_0(
+                &yes,
+                interval.as_millis() as i32,
```

Comment:
```suggestion
                interval.as_millis().try_into().unwrap(),
```
Actually casting `u64` to `i32` can be dangerous so we should panic in case of overflow.